mlxtend/feature_selection/sequential_feature_selector.py

```diff
@@ -550,6 +565,20 @@ def fit(self, X, y, custom_feature_names=None, groups=None, **fit_params):
                                           X)
                     raise KeyboardInterrupt
 
+                # early stop
+                if self.early_stop_rounds \
+                        and k != k_to_select \
+                        and self.k_features in {'best', 'parsimonious'}:
```

Comment:
Instead of the check here, i would maybe change it to raising a `ValueError` in the top of the function if self.k_features is not in {'best', 'parsimonious'} and self.early_stop_rounds. This way the user is aware, and we only have to perform the check once.

Reply:
Do you prefer having this check on top of `fit` function or during initialization?

Reply:
Ahh, totally lost track and missed your comment! Sorry! Regarding your question, I think `fit` might be better to keep it more consistent with scikit-learn behavior.